src/json-patch-duplex.ts

```diff
@@ -290,11 +290,15 @@ module jsonpatch {
   }
 
   function deepClone(obj:any) {
-    if (typeof obj === "object") {
-      return JSON.parse(JSON.stringify(obj)); //Faster than ES5 clone - http://jsperf.com/deep-cloning-of-objects/5
-    }
-    else {
-      return obj; //no need to clone primitives
+    switch (typeof obj) {
+      case "object":
+        return JSON.parse(JSON.stringify(obj)); //Faster than ES5 clone - http://jsperf.com/deep-cloning-of-objects/5
+
+      case "undefined":
+        return null; //this is how JSON.stringify behaves for array items
```

Comment:
I think it does not, because it is covered in tests. 

Can you think of some particular example where it fails?